concent_api/core/tests/test_core_views.py

```diff
@@ -46,97 +85,176 @@ def test_send_should_return_http_200_if_message_timeout(self):
+                PROVIDER_PRIVATE_KEY,
+                settings.CONCENT_PUBLIC_KEY
+            )
+        )
+
+        response = self.client.post(
+            reverse('core:send'),
+            data                           = dump(message_force_report_computed_task_2, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY),
+            content_type                   = 'application/octet-stream',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
+        )
 
         self.assertEqual(response.status_code, 400)  # pylint: disable=no-member
         self.assertIn('error', response.json().keys())
 
     @freeze_time("2017-11-17 10:00:00")
     def test_send_should_return_http_400_if_task_id_already_use(self):
-        response_202 = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
+        response_202 = self.client.post(
+            reverse('core:send'),
+            data                           = dump(self.message_force_report_computed_task, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY),
+            content_type                   = 'application/octet-stream',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
+        )
 
         self.assertIsInstance(response_202, HttpResponse)
         self.assertEqual(response_202.status_code, 202)
 
-        response_400 = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
+        response_400 = self.client.post(
+            reverse('core:send'),
+            data                           = dump(self.message_force_report_computed_task, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY),
+            content_type                   = 'application/octet-stream',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
+        )
 
         self.assertIsInstance(response_400, JsonResponse)
         self.assertEqual(response_400.status_code, 400)  # pylint: disable=no-member
         self.assertIn('error', response_400.json().keys())
 
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_should_return_http_400_if_get_invalid_type_of_message(self):
+
+        response_400 = self.client.post(
+            reverse('core:send'),
+            data = dump(self.message_want_to_compute, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY),
+            content_type                   = 'application/octet-stream',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(CONCENT_PUBLIC_KEY).decode('ascii'),
+        )
+
+        self.assertEqual(response_400.status_code, 400)
+        self.assertIn('error', response_400.json().keys())
+        self.assertIn(response_400.json()['error'], 'Invalid golem message type')
```

Comment:
Never test messages like this. You can check that the message is present but the message itself might be different for various reasons like translations. We also don't want the test to fail if someone simply rewords the message.